Cargo.toml

```diff
@@ -49,6 +49,8 @@ members = [
 
 exclude = ["crates/release-automation"]
 
+resolver = "2"
```

Comment:
I think we'd be better off moving to this because it encourages us to treat our features the same way many (basic on observation in discord) hApp devs are using them

Reply:
Does having this in the workspace Cargo.toml actually translate into, say, the holochain crate when it's published to crates.io? I find the relation ship between workspace and crate Cargo.tomls very confusing.

Reply:
I'm learning new things about Cargo's behaviour by the day but my understanding of this is that it only affects how our workspace gets built and is not published. It just means that when somebody installs `holochain` or any other crate from this workspace into their project using resolver 2 like https://github.com/steveeJ-forks/hc-0.1.4-build-issue/blob/main/Cargo.toml#L4 then we've already built with the same feature enablement logic and we shouldn't have surprises caused by this particular Cargo behaviour

Reply:
From what I've seen, using that resolver arose from having issues with sqlite. A dog biting its tail ;-)
I don't fully understand the repercussions of moving to that resolver, but they should be greater for a repo like Holochain because of its size than for a happ.

Reply:
I think there are two distinct issues happening with sqlite. There's this linker issue which is a consistency issue within this repository that we can resolve with this PR. The other is that it's very easy to pollute your wasm builds when using sweetest and `resolver = "2"` fixes that https://github.com/holochain/holochain/issues/2222.

So I think regardless of linker errors and this change, it makes sense for people to use `resolver = "2"`. I've proposed making the scaffolding opinionated about that https://github.com/holochain/scaffolding/pull/88 and so I think it's best we use the same option to reduce differences between how we build and how we expect consumers of these crates to build